app/src/main/java/com/poupa/vinylmusicplayer/preferences/SharedPreferencesExporter.java

```diff
@@ -0,0 +1,84 @@
+package com.poupa.vinylmusicplayer.preferences;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.ParcelFileDescriptor;
+
+import com.google.gson.Gson;
+import com.poupa.vinylmusicplayer.R;
+import com.poupa.vinylmusicplayer.util.OopsHandler;
+import com.poupa.vinylmusicplayer.util.SafeToast;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
+
+public class SharedPreferencesExporter extends AppCompatActivity {
+    private Context context;
+    private static final String FILENAME = "filename";
+    private ActivityResultLauncher<String> exportFilePicker;
+    private SharedPreferences sharedPreferences;
+
+    public static void start(Context context, String filename) {
+        Intent intent = new Intent(context, SharedPreferencesExporter.class);
+        intent.putExtra(FILENAME, filename);
+        context.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.context = this.getApplicationContext();
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this.context);
+        Bundle b = this.getIntent().getExtras();
+        exportFilePicker = registerForActivityResult(new ActivityResultContracts.CreateDocument("text/plain"), result -> {
+            // Unless the selection has been cancelled, create the export file
+            if(result != null) {
+                try {
+                    writeToExportFile(result);
+                } catch (PackageManager.NameNotFoundException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            // Finishes the last activity to return to the settings activity.
+            this.finish();
+        });
+        exportFilePicker.launch(b.getString(FILENAME));
+    }
+
+    private void writeToExportFile(Uri location) throws PackageManager.NameNotFoundException {
+        Gson gson = new Gson();
+        HashMap<String, Object> prefsMap = new HashMap<>(sharedPreferences.getAll());
```

Comment:
Last week I had to use this feature, the hard way :-) My device crashed and I've imported the exported prefs from another device.

Few observations:
- Generally speaking this works.
- I've noticed that after import, the UI color is not consistent between the main screen (following the theme color) and the OS top status bar/bottom nav bar (doesnt follow the theme color). I had to open Settings screen and set a new accent/main color pairs.
- The content of different static playlist contents, since saved in prefs, are exported as well. To me this doesnt make sense for the following reasons:
  - The exported playlist via this feature contains song ID, which are not guaranteed to be stable across different devices. Reimported playlist actually is empty since the imported IDs are orphan.
  - On the new device, the songs are to be copied by the user, and only once this completes it would make sense to reimport the static playlists, i.e. separately from importing prefs. Exporting/importing playlists with `.m3u` files is already possible with the this merged PR - https://github.com/VinylMusicPlayer/VinylMusicPlayer/pull/922
  To not save the static playlists content with the Export operation, after the `SharedPreferences.getAll` we might want to blacklist some of the prefs that should not be exported.

Reply:
Thanks for your suggestion. I've implemented a filter to exclude keys from the export. Is it sufficent to exclude the "SONG_IDS_" prefix for now or can this lead to unwanted side effects?

I don't know if I understand you correctly, that some components aren't colored as intended:
After importing another color the app looks like this in my case:

![grafik](https://github.com/VinylMusicPlayer/VinylMusicPlayer/assets/18245586/3076deb9-a7ab-4466-8a6c-242fac325a4f)

This is how it should look like after importing, isn't it?
From which commit-sha have you installed the app, maybe I can try to reproduce the behavior?

Reply:
> After importing another color the app looks like this in my case:

Your screenshot looks correct (the top and bottom UI shares the same color as the "Settings" title bar). IIRC mine was not on the same color. If I can reproduce this I'll let you know.